source/server.go

```diff
@@ -106,10 +113,10 @@ func (s *Server) recvRecords(stream pb.SourceService_StreamServer) error {
 }
 
 func (s *Server) SendAck(position sdk.Position) error {
-	stream := s.stream.Load()
-	if stream == nil {
-		return fmt.Errorf("no stream is open")
+	for s.stream.Load() == nil {
+		<-s.StreamIsSetCh
```

Comment:
This could block forever if the stream doesn't reopen. We should simultaneously listen to the context from `Ack` and stop blocking if it gets closed.